src/BizHawk.BizInvoke/MemoryBlockLinuxPal.cs

```diff
@@ -43,19 +43,14 @@ public void Dispose()
 
 		private static MemoryProtection ToMemoryProtection(Protection prot)
 		{
-			switch (prot)
+			return prot switch
 			{
-				case Protection.None:
-					return MemoryProtection.None;
-				case Protection.R:
-					return MemoryProtection.Read;
-				case Protection.RW:
-					return MemoryProtection.Read | MemoryProtection.Write;
-				case Protection.RX:
-					return MemoryProtection.Read | MemoryProtection.Execute;
-				default:
-					throw new ArgumentOutOfRangeException(nameof(prot));
-			}
+				Protection.None => MemoryProtection.None,
+				Protection.R => MemoryProtection.Read,
+				Protection.RW => MemoryProtection.Read | MemoryProtection.Write,
+				Protection.RX => MemoryProtection.Read | MemoryProtection.Execute,
+				_ => throw new ArgumentOutOfRangeException(nameof(prot)),
```

Comment:
This and a couple of similar changes are now giving an analyzer error `BHI1005`:
![image](https://github.com/TASEmulators/BizHawk/assets/35152647/a79b2a38-8a9b-48e2-b7d2-053c742c1658)
@YoshiRulz thoughts?

Reply:
I think we were going to get rid of that rule since the compiler generates the `_ => throw IOE` branch. That doesn't mean remove all the manually-written ones, or change them all, but this change at least is reasonable.

Reply:
yeah it looks like this is erroring because of a custom analyzer rule that is no longer necessary/not accurate in this scenario